src/user/user.controller.ts

```diff
@@ -78,6 +153,53 @@ export class UserController {
       if (error instanceof ValidResetPasswordTokenFoundException) {
         throw new PreconditionFailedException(error.message);
       }
+      throw error;
+    }
+  }
+
+  @ApiOperation({ description: 'Rota para editar dados de usuário.' })
+  @HttpCode(HttpStatus.OK)
+  @ApiBearerAuth()
+  @Patch('')
+  async editUser(@Req() req: Request, @Body() data: UserEditDTO) {
```

Comment:
Faltou você tipar o retorno dessa rota com algo assim:

```suggestion
  async editUser(@Req() req: Request, @Body() data: UserEditDTO): Promise<Student | Teacher> {
```